Web/Edubase.Web.UI/Areas/Groups/Views/Group/EditDetails.cshtml

```diff
@@ -140,13 +140,14 @@
 
                     @if (Model.GroupType.OneOfThese(L.MultiacademyTrust, L.SingleacademyTrust))
                     {
+                        TempData["companiesHouseNumber"] = @Model.CompaniesHouseNumber;
                         <div class="govuk-form-group">
                             <label class="govuk-label" for="companiesHouseNumber">
                                 Company number<br />
                                 (also known as Companies House (CH) number)
                             </label>
                             <input type="text" value="@Model.CompaniesHouseNumber" class="govuk-input" disabled="disabled" id="companiesHouseNumber" />
-                            @Html.HiddenFor(x => x.CompaniesHouseNumber)
+                            @* @Html.HiddenFor(x => x.CompaniesHouseNumber) *@
                         </div>
```

Comment:
I do not fully understand the code changes here and how it helps with the access control - maybe we could have a call and talk through it? :) 

I see that the other fields have an access / permission check around it like these examples below -- have we tried an approach like this to control if the input is visually displayed as editable vs disabled/hidden?

It may also be that the backend API and/or permissions within the database need to be updated also.

```cshtml
                    @if (Model.CanUserEditUkprn)
                    {
                        // editable field
                    }
                    else 
                    {
                        // disabled or hidden/"not visible" input
                    }

                    @if (Model.CanUserEditStatus)
                    {
                        // editable field
                    }
                    else 
                    {
                        // disabled or hidden/"not visible" input
                    }

                    @if (Model.CanUserCloseAndMarkAsCreatedInError)
                    {
                        // editable field
                    }
                    else 
                    {
                        // disabled or hidden/"not visible" input
                    }
```

Reply:
Sure Roger, we can have a call about it. It already uses a similar code on the front end 
so the code change is actual in one of those blocks.

Reply:
@Kola-Abolarinwa has kindly spared some time to walk through this, thank you! 

Notes to self / anybody else following along:
- `TempData` is provided by MVC, where values are scoped to the current user's session and the value is removed on the first read of this data
- Potential bug possible if editing multiple groups within multiple tabs (`TempData` is global for the session, not per browser tab) - planned fix for this specific field is to suffix the key with the group`s UID to prevent 

**Once the key has been updated this implementation appears to work as desired to prevent user-supplied values entering into the system for the companies house number field, while also making sure that it is available to submit to the backend API.**

Additional database changes are required for belt+braces, and to address the underlying issue of the wrong permissions being applied to some users within the database (not visible due to the C# code disregarding any database/API permissions and having hard-coded permissions logic within the view layer).

A (much!) broader question is how/whether to address this more broadly than companies house number, and rearchitecting things to have a single source of truth for all authorisation checks rather than having it spread across multiple layers.

FYI @CathLass @ScottDawsonDfE @anthony9853